src/com/dotmarketing/portlets/rules/business/RulesEngine.java

```diff
@@ -3,36 +3,42 @@
 import com.dotcms.enterprise.LicenseUtil;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Ruleable;
 import com.dotmarketing.business.web.WebAPILocator;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.portlets.rules.exception.RuleEngineException;
 import com.dotmarketing.portlets.rules.model.Rule;
 import com.dotmarketing.util.Config;
 import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.WebKeys;
 import com.liferay.portal.model.User;
+
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Set;
+
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 public final class RulesEngine {
 	private static int SLOW_RULE_LOG_MIN=Config.getIntProperty("SLOW_RULE_LOG_MIN", 100);
 
-    public static void fireRules(HttpServletRequest req, HttpServletResponse res, Rule.FireOn fireOn) {
-
+    public static void fireRules(HttpServletRequest req, HttpServletResponse res, Ruleable parent, Rule.FireOn fireOn) {
         //Check for the proper license level, the rules engine is an enterprise feature only
         if ( LicenseUtil.getLevel() < 200 ) {
             return;
         }
-
-        Host host;
-
-        try {
-            host =  WebAPILocator.getHostWebAPI().getCurrentHost(req);
-        } catch (Exception e) {
-            Logger.error(RulesEngine.class, "Unable to retrieve current request host for URI ", e);
-            return;
+        if(req !=null ){
```

Comment:
It probably should be - I was just being wary of all our page generation type, like the timemachine or any unit tests.  We could throw there, np